drivers/bus/isapnp/isapnp.c

```diff
@@ -194,323 +152,295 @@ IsaConvertMemRange32Requirement(
-    {
-        if (!LogDev->Irq[i].Description.Mask)
-            continue;
-
-        FirstDescriptor = TRUE;
-
-        for (j = 0; j < RTL_BITS_OF(LogDev->Irq[i].Description.Mask); j++)
-        {
-            if (!(LogDev->Irq[i].Description.Mask & (1 << j)))
-                continue;
-
-            IsaConvertIrqRequirement(Descriptor++,
-                                     &LogDev->Irq[i].Description,
-                                     j,
-                                     FirstDescriptor);
-
-            if (FirstDescriptor)
-                FirstDescriptor = FALSE;
-        }
-    }
-    for (i = 0; i < RTL_NUMBER_OF(LogDev->Dma); i++)
+    IsFirstAltList = TRUE;
+    AltList = &RequirementsList->List[0];
+    Descriptor = &RequirementsList->List[0].Descriptors[0];
+    Resource = PdoExt->IsaPnpDevice->Resources;
+    while (Resource->Type != ISAPNP_RESOURCE_TYPE_END)
     {
-        if (!LogDev->Dma[i].Description.Mask)
-            continue;
-
-        FirstDescriptor = TRUE;
-
-        for (j = 0; j < RTL_BITS_OF(LogDev->Dma[i].Description.Mask); j++)
+        switch (Resource->Type)
         {
-            if (!(LogDev->Dma[i].Description.Mask & (1 << j)))
-                continue;
-
-            IsaConvertDmaRequirement(Descriptor++,
-                                     &LogDev->Dma[i].Description,
-                                     j,
-                                     FirstDescriptor);
-
-            if (FirstDescriptor)
-                FirstDescriptor = FALSE;
-        }
-    }
-    for (i = 0; i < RTL_NUMBER_OF(LogDev->MemRange); i++)
-    {
-        if (!LogDev->MemRange[i].Description.Length)
-            continue;
+            case ISAPNP_RESOURCE_TYPE_START_DEPENDENT:
+            {
+                if (!IsFirstAltList)
+                {
+                    /* Add room for the fixed descriptors */
+                    AltList->Count += LastFixedDescriptors;
 
-        IsaConvertMemRangeRequirement(Descriptor++,
-                                      &LogDev->MemRange[i].Description);
-    }
-    for (i = 0; i < RTL_NUMBER_OF(LogDev->MemRange32); i++)
-    {
-        if (!LogDev->MemRange32[i].Description.Length)
-            continue;
+                    /* Move on to the next list */
+                    AltList = (PIO_RESOURCE_LIST)(AltList->Descriptors + AltList->Count);
+                    AltList->Version = 1;
+                    AltList->Revision = 1;
 
-        IsaConvertMemRange32Requirement(Descriptor++,
-                                        &LogDev->MemRange32[i].Description);
-    }
-    if (Alternatives)
-    {
-        UCHAR BestConfig[ISAPNP_MAX_ALTERNATIVES];
-        PIO_RESOURCE_LIST AltList = &RequirementsList->List[0];
-        PIO_RESOURCE_LIST NextList = AltList;
+                    /* Propagate the fixed resources to our new list */
+                    RtlCopyMemory(&AltList->Descriptors,
+                                  &RequirementsList->List[0].Descriptors[0],
```

Comment:
Would this be equivalent to `RequirementsList->List[0].Descriptors` ?

Reply:
Done